gui/wxpython/gui_core/pyedit.py

```diff
@@ -681,6 +688,17 @@ def _set_overwrite(self, overwrite):
     def OnSetParameters(self, *args, **kwargs):
         self.controller.OnSetParameters(*args, **kwargs)
 
+    def OnRecentFilesMenu(self, event):
+        id = event.GetId()
+        menu = event.GetEventObject()
+        label = menu.GetLabelText(id)
+        index = int(label.split(' ')[0]) - 1
+        fd = open(self.filehistory.GetHistoryFile(index=index), 'r')
+        try:
+            self.controller.body.SetText(fd.read())
+        finally:
+            fd.close()
```

Comment:
Consider using `with` instead of `try-finally`.

Reply:
I've already fixed it.